components/CodeBlock.tsx

```diff
@@ -25,7 +28,36 @@ export const RawCodeBlock = ({
   language,
   className: extraClassName,
   disablePrefixes,
-}: CodeBlockProps & { className?: string }) => {
+  enableLineRef = false,
+}: CodeBlockProps & { className?: string; enableLineRef?: boolean }) => {
+  const [hashValue, setHashValue] = useState("");
+  useEffect(() => {
+    Router.events.on("hashChangeComplete", (url: any) => {
+      setHashValue(url.slice(url.indexOf("#")));
+    });
+    const { hash } = location;
+    setHashValue(hash);
+    return () => {
+      Router.events.off("hashChangeComplete", () => {});
+    };
+  }, []);
```

Comment:
Can't you use the `useRouter` hook for this? Might not be possible, but seems cleaner.

Reply:
Okay sure

Reply:
Oh yeah you're right. Looks good then :-)